lockbox2/programs/liquidity_lockbox/src/lib.rs

```diff
@@ -533,16 +526,21 @@ pub struct DepositPositionForLiquidity<'info> {
   #[account(mut, constraint = token_vault_b.key() == whirlpool.token_vault_b)]
   pub token_vault_b: Box<Account<'info, TokenAccount>>,
 
-  #[account(mut, has_one = whirlpool, constraint = tick_array_lower.key() != tick_array_upper.key())]
+  #[account(mut, has_one = whirlpool,
+    constraint = tick_array_lower.key() != tick_array_upper.key(),
+    constraint = tick_array_lower.to_account_info().owner == &whirlpool_program.key()
+  )]
   pub tick_array_lower: AccountLoader<'info, TickArray>,
-  #[account(mut, has_one = whirlpool)]
+  #[account(mut, has_one = whirlpool,
+    constraint = tick_array_upper.to_account_info().owner == &whirlpool_program.key()
+  )]
   pub tick_array_upper: AccountLoader<'info, TickArray>,
 
   #[account(mut, address = lockbox.bridged_token_mint)]
   pub bridged_token_mint: Box<Account<'info, Mint>>,
   #[account(mut,
     constraint = bridged_token_account.mint == lockbox.bridged_token_mint,
-    constraint = bridged_token_mint.key() == lockbox.bridged_token_mint,
+    constraint = bridged_token_account.mint == bridged_token_mint.key(),
     constraint = signer.key == &bridged_token_account.owner,
```

Comment:
Switching context of verified accounts to correspond to actual declared accounts.